ui/src/shared/components/XYContainer.tsx

```diff
@@ -49,15 +62,24 @@ const XYContainer: FunctionComponent<Props> = ({
       },
     },
   },
+  onSetXColumn,
+  onSetYColumn,
 }) => {
   const {table, fluxGroupKeyUnion} = useMemo(
     () => fromFlux(files.join('\n\n')),
     [files]
   )
 
-  // Eventually these will be configurable in the line graph options UI
-  const xColumn = chooseXColumn(table)
-  const yColumn = chooseYColumn(table)
+  useEffect(() => {
+    if (!xColumn) {
+      xColumn = chooseXColumn(table)
+      onSetXColumn(xColumn)
+    }
+    if (!yColumn) {
+      yColumn = chooseYColumn(table)
+      onSetYColumn(yColumn)
+    }
+  })
 
```

Comment:
(same comment about ensuring the `xColumn` and `yColumn` selections actually exist in the `table`)